### PR TITLE
chore: send connection config to router transform

### PR DESCRIPTION
**Pull request by Sidddddarth**
# Description

Rudder server now sends `connection config` to router transform with each job.
Due to this being pertinent only in case of sources jobs, we have a check to drop job only if it's a sources job.

## Linear Ticket

[Resolves PIPE-1222](https://linear.app/rudderstack/issue/PIPE-1222/read-connection-config-and-use-it-in-relevant-places)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by ktgowtham** [comment]
Adding @koladilip . He wanted to review the PR.

**Comment by koladilip** [comment]
@Sidddddarth you will be raising for process or proxy later?

File: router/types/types.go [hunk 1]
```diff
@@ -18,6 +18,16 @@ type RouterJobT struct {
...
 	JobMetadata JobMetadataT               `json:"metadata"`
 	Destination backendconfig.DestinationT `json:"destination"`
+	Connection  backendconfig.Connection   `json:"connection"`
+}
+
+type SourceDest struct {
+	SourceID, DestinationID string
+}
+
+type Connection struct {
```
Comment (koladilip) [inline router/types/types.go:28]: where is this used?

Comment (Sidddddarth) [reply router/types/types.go:28]: It contains the connection config - so used to populate wherever needed
I've created this new struct just to have the `connectionID` here as well - could help with stats or maybe other uses in the future.

**Review by koladilip — APPROVED** [review]

**Review by ktgowtham — APPROVED** [review]
LGTM

**Comment by koladilip** [comment]
we tested this @Sidddddarth

File: router/types/types.go (hunk 1 above)
Comment (cisse21) [inline router/types/types.go:28]:
```suggestion
type ConnectionWithID struct {
```

File: router/types/types.go [hunk 2]
```diff
@@ -18,6 +18,16 @@ type RouterJobT struct {
...
+	ConnectionID string
+	ConnectionT  backendconfig.Connection
```
Comment (cisse21) [inline router/types/types.go:30]:
```suggestion
	Connection  backendconfig.Connection
```

**Review by cisse21 — APPROVED** [review]
Approving with a nitpick

**Pull request merged**